src/main/java/io/supertokens/session/accessToken/AccessTokenSigningKey.java

```diff
@@ -90,29 +96,80 @@ synchronized void removeKeyFromMemoryIfItHasNotChanged(List<KeyInfo> oldKeyInfo)
         }
     }
 
-    public synchronized List<KeyInfo> getKey()
+    public synchronized void transferLegacyKeyToNewTable()
             throws StorageQueryException, StorageTransactionLogicException {
+        Storage storage = StorageLayer.getSessionStorage(main);
+
+        if (storage.getType() == STORAGE_TYPE.SQL) {
+            SessionSQLStorage sqlStorage = (SessionSQLStorage) storage;
+
+            // start transaction
+            sqlStorage.startTransaction(con -> {
+                KeyValueInfo legacyKey = sqlStorage.getLegacyAccessTokenSigningKey_Transaction(con);
+
+                if (legacyKey != null) {
+                    sqlStorage.addAccessTokenSigningKey_Transaction(con, legacyKey);
+                    sqlStorage.removeLegacyAccessTokenSigningKey_Transaction(con);
+                }
+                return legacyKey;
+            });
+        } else {
+            while (true) {
+                SessionNoSQLStorage_1 noSQLStorage = (SessionNoSQLStorage_1) storage;
+                KeyValueInfoWithLastUpdated legacyKey = noSQLStorage.getLegacyAccessTokenSigningKey_Transaction();
+
+                if (legacyKey != null) {
+                    // We should only get here once, after an upgrade.
+                    // We avoid adding duplicates by enforcing that a legacy key can only ever be
+                    // the first one in the new table.
```

Comment:
How are we enforcing this? By removing the legacy key post update right? Or some other trick?

Reply:
We are setting lastCreated to 0, which means that there is no other key.

Reply:
It's actually better to set it to -1, as it better signals the intent.

Reply:
You can make the plugin interface function take a `Long`. This way, you can give the function `null` instead of `-1` or `0`